governator-core/src/main/java/com/netflix/governator/lifecycle/DefaultClasspathScanner.java

```diff
@@ -0,0 +1,166 @@
+import com.google.common.collect.Sets;
+
+/**
+ * Utility to find annotated classes
+ */
+public class DefaultClasspathScanner extends ClasspathScanner
+{
+    private static final Logger log = LoggerFactory.getLogger(ClasspathScanner.class);
+    protected final ClassLoader classLoader;
+    
+    private final Set<Class<?>> classes;
+    private final Set<Constructor> constructors;
+    private final Set<Method> methods;
+    private final Set<Field> fields;
+
+    /**
+     * @param basePackages list of packages to search (recursively)
+     * @param annotations class annotations to search for
+     */
+    public DefaultClasspathScanner(Collection<String> basePackages, Collection<Class<? extends Annotation>> annotations)
+    {
+        this(basePackages, annotations, Thread.currentThread().getContextClassLoader());
+    }
+
+    /**
+     * @param basePackages list of packages to search (recursively)
+     * @param annotations class annotations to search for
+     * @param classLoader ClassLoader containing the classes to be scanned
+     */
+    public DefaultClasspathScanner(Collection<String> basePackages, Collection<Class<? extends Annotation>> annotations, final ClassLoader classLoader) 
+    {
+        Preconditions.checkNotNull(annotations, "additionalAnnotations cannot be null");
+        Preconditions.checkNotNull(classLoader, "classLoader cannot be null");
+
+        log.debug("Starting classpath scanning...");
+        this.classLoader = classLoader;
+
+        Set<Class<?>>       localClasses = Sets.newHashSet();
+        Set<Constructor>    localConstructors = Sets.newHashSet();
+        Set<Method>         localMethods = Sets.newHashSet();
+        Set<Field>          localFields = Sets.newHashSet();
+
+        doScanning(basePackages, annotations, localClasses, localConstructors, localMethods, localFields);
+
+        classes = ImmutableSet.copyOf(localClasses);
+        constructors = ImmutableSet.copyOf(localConstructors);
+        methods = ImmutableSet.copyOf(localMethods);
+        fields = ImmutableSet.copyOf(localFields);
+
+        log.debug("Classpath scanning done");
+    }
+
+    /**
+     * @return the found classes
+     */
+    @Override
+    public Set<Class<?>> getClasses()
+    {
+        return classes;
+    }
+
+    @Override
+    public Set<Constructor> getConstructors()
+    {
+        return constructors;
+    }
+
+    @Override
+    public Set<Method> getMethods()
+    {
+        return methods;
+    }
+
+    @Override
+    public Set<Field> getFields()
+    {
+        return fields;
+    }
+
+    protected void doScanning(Collection<String> basePackages, Collection<Class<? extends Annotation>> annotations, Set<Class<?>> localClasses, Set<Constructor> localConstructors, Set<Method> localMethods, Set<Field> localFields)
+    {
+        if ( basePackages.size() == 0 )
+        {
+            log.warn("No base packages specified - no classpath scanning will be done");
```

Comment:
As I typically want to disable classpath scanning, I would prefer not getting a warning about it. That said, it looks like this change splits out the interface so I can easily pass in a no-op implementation.